src/custom/state/claim/hooks/index.ts

```diff
@@ -385,6 +385,61 @@ export function useUsdcPrice(): string | null {
 
 type VCowPriceFnNames = 'nativeTokenPrice' | 'gnoPrice' | 'usdcPrice'
 
+/**
+ * Hook that calls specific method on the vCow contract defined on the VCowMethods type
+ * and returns the value in the form of JSBI.BigInt
+ *
+ * @param method method that will be called on the vCowContract
+ */
+type VCowMethods = 'swappableBalanceOf' | 'balanceOf'
+
+export function _useFetchVCowValue(method: VCowMethods) {
+  const vCowContract = useVCowContract()
+  const { chainId, account } = useActiveWeb3React()
+
+  const [value, setValue] = useState<null | JSBI>(null)
+
+  useEffect(() => {
+    if (!chainId || !vCowContract || !account) {
+      return
+    }
+
+    vCowContract[method](account)
+      .then((result) => {
+        setValue(JSBI.BigInt(result))
+      })
+      .catch((error: Error) => {
+        console.log(`[vCowContract] ${method} fetch failed`, error)
```

Comment:
should we expose the error out of the hook or sth instead of silence it?

Reply:
Not relevant anymore after latest update